pkg/controller/iam/managed_test.go

```diff
@@ -0,0 +1,797 @@
+			}
+		})
+	}
+
+}
+
+func TestObserve(t *testing.T) {
+	type args struct {
+		ctx context.Context
+		mg  resource.Managed
+	}
+	type want struct {
+		mg          resource.Managed
+		observation managed.ExternalObservation
+		err         error
+	}
+
+	cases := map[string]struct {
+		handler http.Handler
+		args    args
+		want    want
+	}{
+		"ObservedAccountGot": {
+			handler: http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				_ = r.Body.Close()
+				if diff := cmp.Diff(http.MethodGet, r.Method); diff != "" {
+					t.Errorf("r: -want, +got:\n%s", diff)
+				}
+				w.WriteHeader(http.StatusOK)
+				sa := &iamv1.ServiceAccount{
+					Name:        fqName,
+					UniqueId:    uniqueID,
+					Email:       accountEmail,
+					DisplayName: displayName,
+				}
+				_ = json.NewEncoder(w).Encode(sa)
+			}),
+			args: args{
+				ctx: context.Background(),
+				mg: serviceAccount(
+					withName(fqName),
+					withExternalNameAnnotation(fqName),
+				),
+			},
+			want: want{
+				mg: serviceAccount(
+					withName(fqName),
+					withUniqueID(uniqueID),
+					withEmail(accountEmail),
+					withDisplayName(displayName),
+					withExternalNameAnnotation(fqName),
+					withDisabled(false)),
+				observation: managed.ExternalObservation{
+					ResourceExists:    true,
+					ResourceUpToDate:  true,
+					ConnectionDetails: managed.ConnectionDetails{},
+				},
+			},
+		},
+		"ObservedServiceAccountDoesNotExist": {
+			handler: http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				r.Body.Close()
+				w.WriteHeader(http.StatusNotFound)
+			}),
+			args: args{
+				ctx: context.Background(),
+				mg:  serviceAccount(),
+			},
+			want: want{
+				mg:          serviceAccount(),
+				observation: managed.ExternalObservation{ResourceExists: false},
+			},
+		},
+		"NotServiceAccount": {
+			args: args{
+				ctx: context.Background(),
+				mg:  &strange{},
+			},
+			want: want{
+				mg:  &strange{},
+				err: errors.New(errNotServiceAccount),
+			},
+		},
+	}
+
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			server := httptest.NewServer(tc.handler)
+			defer server.Close()
+			s, _ := iamv1.NewService(context.Background(), option.WithEndpoint(server.URL), option.WithoutAuthentication())
+			serviceAccounts := iamv1.NewProjectsService(s).ServiceAccounts
+			rrn := NewRelativeResourceNamer("perfect-project")
+			e := &external{serviceAccounts: serviceAccounts, rrn: rrn}
+			obs, err := e.Observe(context.Background(), tc.args.mg)
+
+			if err != nil {
+				if tc.want.err != nil {
+					// we expected a different error than we got
+					if diff := cmp.Diff(tc.want.err.Error(), err.Error()); diff != "" {
+						t.Errorf("Observe(...): want error string != got error string:\n%s", diff)
+					}
+				} else {
+					t.Errorf("Observe(...): unexpected error %s", err)
+				}
+			} else {
+				if tc.want.err != nil {
+					t.Errorf("Observe(...) want error %s got nil", tc.want.err)
+				}
+			}
```

Comment:
Could we use `cmp.Diff(tc.want.err, err, test.EquateErrors())` here like we do in the other tests? If not, could you include a comment explaining why?